gemini/evaluation/enhancing_quality_and_explainability_with_rapideval.ipynb

```diff
@@ -0,0 +1,665 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "e761e89b9065"
+   },
+   "outputs": [],
+   "source": [
+    "generation_model = GenerativeModel(\"gemini-1.5-pro-001\")\n",
+    "generation_config = GenerationConfig(\n",
+    "    temperature=0.4, max_output_tokens=512, candidate_count=1\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "8669160b92b9"
+   },
+   "source": [
+    "### Prompt gemini\n",
+    "Now we prompt gemini to generate multiple slightly de-corelated responses based on the above config. Multiple Responses can also be generated in single call.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "3fb5e3ae9faf"
+   },
+   "outputs": [],
+   "source": [
+    "instruction_qa = \"Please answer the following question based on the context provided. Question: what is the correct process of fixing your tires?\"\n",
+    "context_qa = (\n",
+    "    \"Context:\\n\"\n",
+    "    + \"the world is a magical place and fixing tires is one of those magical tasks. According to the Administration and Association (TIA), the only method to properly repair a tire puncture is to fill the injury with a repair stem and back the stem with a repair patch. This is commonly known as a combination repair or a patch/plug repair.\"\n",
+    ")\n",
+    "prompt_qa = instruction_qa + \"\\n\" + context_qa + \"\\n\\nAnswer:\\n\"\n",
+    "responses = list()\n",
+    "num_responses = 5\n",
+    "for i in range(num_responses):\n",
+    "    responses.append(\n",
+    "        generation_model.generate_content(\n",
+    "            contents=prompt_qa,\n",
+    "            generation_config=generation_config,\n",
+    "        ).text\n",
+    "    )\n",
+    "prompt_qa"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "c54c44f3f0b5"
+   },
+   "source": [
+    "Here we use the rank_responses function to fetch the best selected response as well as its associated quality metrics.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "20230a87d5c1"
+   },
+   "outputs": [],
+   "source": [
+    "best_response, metrics = rank_responses(instruction_qa, context_qa, responses)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "6b5b63a55144"
+   },
+   "source": [
+    "Now we print the various generated responses:\n",
+    "1. the various responses which were generated by gemini\n",
+    "2. The best performing response amongst them\n",
+    "3. Its associated pointwise quality metrics and explanation.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "7ba246f23c1f"
+   },
+   "outputs": [],
+   "source": [
+    "for i in range(len(responses)):\n",
+    "    print(\"Response no. {}: \\n {}\".format(i + 1, responses[i]))"
+   ]
```

Comment:
You can iterate over this with `enumerate(responses, start=1)` to print the index starting at 1.